docs/pytorch_project_convertor/API_docs/ops/torch.gather.md

```diff
@@ -66,3 +66,34 @@ paddle.gather(t, paddle.to_tensor([1, 0]), 1)
 #        [[2, 1],
 #         [4, 3]])
 ```
+
+### 组合实现
+
+```python
+def paddle_gather(x, dim, index):
+    index_shape = index.shape
+    index_flatten = index.flatten()
+    if dim < 0:
+        dim = len(x.shape) + dim
+    nd_index = []
+    for k in range(len(x.shape)):
+        if k == dim:
+            nd_index.append(index_flatten)
+        else:
+            reshape_shape = [1] * len(x.shape)
+            reshape_shape[k] = x.shape[k]
+            dim_index = paddle.expand(paddle.reshape(paddle.arange(
+                x.shape[k], dtype=index.dtype), reshape_shape), index_shape).flatten()
```

Comment:
> 代码风格需格式化一下，建议commit之前，运行X2Paddle目录下的commit-prepare.sh
> 
> ```python
> sh commit-prepare.sh
> ```

收到，这是我在网上找到的，正好用到，试过是没问题的。发现x2paddle中没有列出所以想加上。明天我会根据要求完成。

Reply:
请问现在修改后，运行了`sh commit-prepare.sh`，其中有个`Failed`，然后就push不了好像
```shell
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
yapf.................................................(no files to check)Skipped
Check for merge conflicts................................................Passed
Check for broken symlinks................................................Passed
Fix End of Files.........................................................Failed
- hook id: end-of-file-fixer
- exit code: 1
- files were modified by this hook

Fixing docs/pytorch_project_convertor/API_docs/ops/torch.gather.md

Trim Trailing Whitespace.................................................Passed
Detect Private Key.......................................................Passed
Check for broken symlinks................................................Passed
Check for added large files..............................................Passed
copyright_checker....................................(no files to check)Skipped
```